features/steps/user_api_steps.py

```diff
@@ -110,13 +105,19 @@ def step_impl(context):
     payload = {
         'pod_id': context.analyzer_job_id
     }
+
+    analyzer_job_id = context.response['pod_id']
     r = requests.get(
-        f'{context.endpoint_url}/log', params=payload)
+        f'{context.endpoint_url}/log/{analyzer_job_id}')
 
     assert_that(r.status_code, equal_to(HTTPStatus.OK))
     assert_that(r.headers['content-type'], equal_to('application/json'))
 
     analyzer_job_log = r.json()
     assert_that(analyzer_job_log, not_none)
+    log_data = r.json()
+    assert_that(log_data, not_none)
+    document_id = log_data['pod_id']
+    assert_that(document_id, not_none)
 
-    assert_that(analyzer_job_log['pod_id'], equal_to(context.analyzer_job_id))
+    context.latest_doc_id = document_id
```

Comment:
Why are you assigning here? The step itself does not mention you are changing in the context.